packages/components/src/Actions/ActionFile/ActionFile.component.js

```diff
@@ -0,0 +1,137 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { OverlayTrigger } from 'react-bootstrap';
+import classNames from 'classnames';
+import uuid from 'uuid';
+
+import TooltipTrigger from '../../TooltipTrigger';
+import CircularProgress from '../../CircularProgress';
+
+import Icon from '../../Icon';
+import theme from './ActionFile.scss';
+
+const LEFT = 'left';
+const RIGHT = 'right';
+
+function getIcon({ icon, iconTransform, inProgress }) {
+	if (inProgress) {
+		return <CircularProgress size="small" key="icon" />;
+	}
+
+	if (icon) {
+		return <Icon name={icon} transform={iconTransform} key="icon" />;
+	}
+
+	return null;
+}
+getIcon.propTypes = {
+	icon: PropTypes.string,
+	iconTransform: PropTypes.string,
+	inProgress: PropTypes.bool,
+};
+
+function getLabel({ hideLabel, label }) {
+	if (hideLabel) {
+		return null;
+	}
+	return <span key="label">{label}</span>;
+}
+
+getLabel.propTypes = {
+	label: PropTypes.string,
+	hideLabel: PropTypes.bool,
+};
+
+function adjustContentPlacement(icon, label, iconPosition) {
+	if (iconPosition === RIGHT) {
+		return [label, icon];
+	}
+	return [icon, label];
+}
```

Comment:
Can't you reuse the same array but reverse it if `iconPosition === RIGHT`?

Reply:
i am not sure to see any added value to your proposal.